src/main/java/net/sf/jabref/gui/importer/ImportFormats.java

```diff
@@ -84,64 +59,56 @@ public static AbstractAction getImportAction(JabRefFrame frame, boolean openInNe
         class ImportAction extends MnemonicAwareAction {
 
             private final JabRefFrame frame;
-            private final boolean openInNew;
-
+            private final boolean newDatabase;
 
-            public ImportAction(JabRefFrame frame, boolean openInNew) {
+            public ImportAction(JabRefFrame frame, boolean newDatabase) {
                 this.frame = frame;
-                this.openInNew = openInNew;
-
-                putValue(Action.NAME, openInNew ? Localization.menuTitle("Import into new database") : Localization
-                        .menuTitle("Import into current database"));
-                putValue(Action.ACCELERATOR_KEY,
-                        openInNew ? Globals.getKeyPrefs().getKey(KeyBinding.IMPORT_INTO_NEW_DATABASE) : Globals
-                                .getKeyPrefs().getKey(KeyBinding.IMPORT_INTO_CURRENT_DATABASE));
+                this.newDatabase = newDatabase;
+
+                if (newDatabase) {
+                    putValue(Action.NAME, Localization.menuTitle("Import into new database"));
+                    putValue(Action.ACCELERATOR_KEY, Globals.getKeyPrefs().getKey(KeyBinding.IMPORT_INTO_NEW_DATABASE));
+                } else {
+                    putValue(Action.NAME, Localization.menuTitle("Import into current database"));
+                    putValue(Action.ACCELERATOR_KEY, Globals.getKeyPrefs().getKey(KeyBinding.IMPORT_INTO_CURRENT_DATABASE));
+                }
             }
 
             @Override
             public void actionPerformed(ActionEvent e) {
-                JFileChooser fileChooser = createImportFileChooser(
-                        Globals.prefs.get(JabRefPreferences.IMPORT_WORKING_DIRECTORY));
-                int result = fileChooser.showOpenDialog(frame);
-
-                if (result != JFileChooser.APPROVE_OPTION) {
-                    return;
-                }
-
-                File file = fileChooser.getSelectedFile();
-                if (file == null) {
-                    return;
-                }
-
-                FileFilter ff = fileChooser.getFileFilter();
-                ImportFormat format = null;
-                if (ff instanceof ImportFileFilter) {
-                    format = ((ImportFileFilter) ff).getImportFormat();
-                }
-
-                try {
-                    if (!file.exists()) {
-                        // Warn that the file doesn't exists:
-                        JOptionPane.showMessageDialog(frame,
-                                Localization.lang("File not found") + ": '" + file.getName() + "'.",
-                                Localization.lang("Import"), JOptionPane.ERROR_MESSAGE);
-                        return;
+                SortedSet<ImportFormat> importers = Globals.IMPORT_FORMAT_READER.getImportFormats();
+                List<FileExtensions> extensions = importers.stream().map(p -> p.getExtensions()).collect(Collectors.toList());
```

Comment:
Not sure if I get the point of your discussion, but in general I support the idea that FileDialog should support a way to set the extensions and file filter based on import formats (because this should happen quite often).
What about adding a `static FileDialog createDialogForImport(frame, workingDir, Collection<ImportFormat>)` which contains most of the initialization code here.